sources/engine/Xenko.Graphics/Direct3D/Texture.Direct3D.cs

```
@@ -154,7 +154,34 @@ private void InitializeFromImpl(DataBox[] dataBoxes = null)
             NativeShaderResourceView = GetShaderResourceView(ViewType, ArraySlice, MipLevel);
             NativeUnorderedAccessView = GetUnorderedAccessView(ViewType, ArraySlice, MipLevel);
             NativeRenderTargetView = GetRenderTargetView(ViewType, ArraySlice, MipLevel);
-            NativeDepthStencilView = GetDepthStencilView(out HasStencil);           
+            NativeDepthStencilView = GetDepthStencilView(out HasStencil);
+
+            try
+            {
+                switch (textureDescription.Options)
+                {
+                    case TextureOptions.None:
+                        SharedHandle = IntPtr.Zero;
+                        break;
+                    case TextureOptions.Shared:
+                        var sharedResource = NativeDeviceChild.QueryInterface<SharpDX.DXGI.Resource>();
+                        SharedHandle = sharedResource.SharedHandle;
+                        break;
+                    case TextureOptions.SharedNthandle:
+                        var sharedResource1 = NativeDeviceChild.QueryInterface<SharpDX.DXGI.Resource1>();
+                        var uniqueName = "Xenko:" + Guid.NewGuid().ToString();
+                        SharedHandle = sharedResource1.CreateSharedHandle(uniqueName, SharpDX.DXGI.SharedResourceFlags.Write);
+                        SharedNtHandleName = uniqueName;
+                        break;
+                    default:
+                        SharedHandle = IntPtr.Zero;
+                        break;
+                }
+            }
+            catch
```

If possible, try to avoid empty `catch` clause. It is better to crash and detect the issue than silently swallowing it.